README.md

```diff
@@ -1,2 +1,97 @@
 # o2r-guestlister
-An OAuth server implementation to allow offline login with o2r-bouncer 
+
+A Node.js OAuth2 server implementation to allow offline login with o2r-bouncer as part of the [o2r reference-implementation](https://github.com/o2r-project/reference-implementation). For more inforamtion about OAuth2 see the OAUth2 [documentation](https://oauth.net/2/).
+
+Based on the repository https://github.com/gerges-beshay/oauth2orize-examples and the underlying OAuth2 server implementation [oauth2orize](https://www.npmjs.com/package/oauth2orize).
+
+Requirements:
+
+- nodejs `>= 6.2`
+- npm
+
+## Endpoints
+
+The Oauth2 server implementation allows trying out the o2r reference-implementation without going through the [ORCID](https://orcid.org/) app registration. It implements the following endpoints:
+
+* `/oauth/authorize` Starts an authorization request granting an authorization code.
+* `/login` User login: Allows to chose between three different uses with basic, advanced and admin rights.
+* `/oauth/token` Exchange an authorization code for an access token.
+
+To mimic the ORCID OAuth2 implementation the demo server submits the `username` and `ORCID` ID as custom parameters in the response to the access token request.
+
+## Demo data
+
+The service creates three test users when starting the service:
```

Comment:
Add a link to http://o2r.info/o2r-web-api/user/#user-levels

Reply:
done